lib/util/search/address_suggestion_manager.dart

```diff
@@ -49,6 +50,16 @@ class AddressSuggestionManager {
     Iterable<Map<String, dynamic>> suggestions = data.map((suggestion) => jsonDecode(suggestion));
     _historySuggestions =
         suggestions.map((suggestion) => AddressSuggestion.fromJson(suggestion, fromHistory: true)).toList();
+
+    _historySuggestionsLoaded = true;
+  }
+
+  void reloadHistorySuggestions() {
+    if (_historySuggestionsLoaded) {
```

Comment:
Is this necessary? Does this throw an error if `_historySuggestions` is empty?

Reply:
yes because it's late... of course we could make the field nullable instead, but that's not really what this is for

Reply:
I would've suggested to initialize it empty, I think that would be functionally equal wouldn't it? Because then you can just call clear without worrying. I understand that semantically you could separate a "real" empty list from one that is not yet loaded, but this is the only point where that case is relevant I think

Reply:
The thing is: When you open the app, the suggestions are filled with the correct value immediately. Isn't that exactly what `late` was made for?

Reply:
Mh okay, I guess, I don't know what the convention would be here, we can leave it

Reply:
I'll change it :D Less code